nep-11.mediawiki

```diff
@@ -0,0 +1,259 @@
+
+==Specification==
+
+===Common methods===
+
+====symbol====
+
+<pre>
+public static string symbol()
+</pre>
+
+Returns a short string symbol of the token managed in this contract. e.g. <code>"MNFT"</code>.
+This symbol SHOULD be short (3-8 characters is recommended), with no whitespace characters or new-lines and SHOULD be limited to the uppercase latin alphabet (i.e. the 26 letters used in English).
+
+This method MUST always return the same value every time it is invoked.
+
+====decimals====
+
+<pre>
+public static byte decimals()
+</pre>
+
+Returns the number of decimals used by the token - e.g. <code>8</code>, means to divide the token amount by <code>100,000,000</code> to get its user representation.
+
+If the token managed in this contract is indivisible, the function SHOULD return <code>0</code>.
+
+If this method returns <code>0</code>, the "Non-divisible NFT methods" must be implemented.
+Otherwise, the "Divisible NFT methods" must be implemented.
+
+This method MUST always return the same value every time it is invoked.
+
+====totalSupply====
+
+<pre>
+public static BigInteger totalSupply()
+</pre>
+
+Returns the total token supply currently in the system.
+
+====balanceOf====
+
+<pre>
+public static BigInteger balanceOf(byte[] owner)
+</pre>
+
+Returns the total amount of NFTs owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+====tokensOf====
+
+<pre>
+public static iterator tokensOf(byte[] owner)
+</pre>
+
+Returns an <code>iterator</code> that contains all of the token ids owned by the specified address.
+
+The parameter <code>owner</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+===Non-divisible NFT methods===
+
+====transfer====
+
+<pre>
+public static bool transfer(byte[] to, byte[] tokenid)
+</pre>
+
+It transfers the token with id <code>tokenid</code> to address <code>to</code>.
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The parameter <code>to</code> SHOULD be a 20-byte address.
+If not, this method SHOULD <code>throw</code> an exception.
+
+The function SHOULD return <code>false</code> if the token that will be transferred has more than one owner.
+
+If the method succeeds, it MUST fire the <code>Transfer</code> event, and MUST return <code>true</code>, even if the token is sent to the owner.
+
+The function SHOULD check whether the owner address equals the caller contract hash.
+If so, the transfer SHOULD be processed; If not, the function SHOULD use the SYSCALL <code>Neo.Runtime.CheckWitness</code> to verify the transfer.
+
+If the <code>to</code> address is a deployed contract, the function SHOULD check the <code>payable</code> flag of this contract to decide whether it should transfer the tokens to this contract.
```

Comment:
We no longer have `payable`. Probably we need something similar to NEP-17 `onPayment`.

Reply:
Removed.

Reply:
`onNEP11Payment`? Does it have scenarios? I think it is unlikely that the NFT will be sent to the contract by mistake.

Reply:
@erikzhang I agree that it's very unlikely that NFT will be sent to the contract by mistake.... but, in my opinion, this is a feature we can't imagine use cases for now, but I'm sure people (contract devs) would like to have that for convenience.

My vote is to add it.

Reply:
Added `onNEP11Payment`.